sql/analyzer/analyzer.go

```diff
@@ -85,6 +97,53 @@ func (ab *Builder) AddPostValidationRule(name string, fn RuleFunc) *Builder {
 	return ab
 }
 
+func duplicateRulesWithout(rules []Rule, excludedRuleName string) []Rule {
+	newRules := make([]Rule, 0, len(rules))
+
+	for _, rule := range rules {
+		if rule.Name != excludedRuleName {
+			newRules = append(newRules, rule)
+		}
+	}
+
+	return newRules
+}
+
+// RemoveOnceBeforeRule removes a default rule from the analyzer which would occur before other rules
+func (ab *Builder) RemoveOnceBeforeRule(name string) *Builder {
```

Comment:
Think I'd probably rather have a generic RemoveRule that goes through all the batches automatically (and enforce that rule names are distinct)